### PR TITLE
Update environment variables

### PR DESCRIPTION
**Pull request by mhoeger**
Resolves https://github.com/Azure/azure-functions-nodejs-worker/issues/174

### CODE REVIEW

File: src/WorkerChannel.ts [hunk 1]
```diff
@@ -241,6 +221,41 @@ export class WorkerChannel implements IWorkerChannel {
    * Environment variables from the current process
    */ 
   public functionEnvironmentReloadRequest(requestId: string, msg: rpc.IFunctionEnvironmentReloadRequest): void {
-    // Not yet implementeds
+    // Add environment variables from incoming
+    let numVariables = (msg.environmentVariables && Object.keys(msg.environmentVariables).length) || 0;
+    systemLog(`Reloading environment variables. Found ${numVariables} variables to reload.`);
```
Comment (pragnagopa) [inline src/WorkerChannel.ts]: Add test with a test list and null/empty list.

File: src/WorkerChannel.ts [hunk 2]
```diff
@@ -98,28 +96,21 @@ export class WorkerChannel implements IWorkerChannel {
...
-        status: Status.Success
-      };
-
+      let err, errorMessage;
       try {
         this._functionLoader.load(msg.functionId, msg.metadata);
       }
       catch(exception) {
-        let errorMessage = `Worker was unable to load function ${msg.metadata.name}: '${exception}'`;
+        errorMessage = `Worker was unable to load function ${msg.metadata.name}: '${exception}'`;
```
Comment (pragnagopa) [inline src/WorkerChannel.ts]: Add unit test that verifies function load exceptinn.

File: test/WorkerChannelTests.ts [hunk 3]
```diff
@@ -51,6 +52,83 @@ describe('WorkerChannel', () => {
...
+        requestId: 'id',
+        functionEnvironmentReloadRequest: null
+      });
+    }).to.not.throw();
+
+    expect(() => {
+      stream.write({
+        requestId: 'id',
+        functionEnvironmentReloadRequest: {
+          environmentVariables: null
```
Comment (pragnagopa) [inline test/WorkerChannelTests.ts]: Also verify process.env is empty
